auth/auth-server/src/server/api_handlers/order_book.rs

```diff
@@ -1,39 +1,140 @@
 //! Orderbook endpoint handlers
 use bytes::Bytes;
+use futures_util::future;
 use http::{HeaderMap, Method, StatusCode};
+use renegade_api::http::order_book::{GetDepthForAllPairsResponse, PriceAndDepth};
+use renegade_common::types::token::Token;
+use tokio::task::JoinHandle;
 use tracing::instrument;
+use uuid::Uuid;
 use warp::reject::Rejection;
 
 use super::{Server, log_unsuccessful_relayer_request};
+use crate::error::AuthServerError;
+use crate::http_utils::request_response::overwrite_response_body;
 use crate::server::api_handlers::external_match::BytesResponse;
 use crate::telemetry::helpers::record_relayer_request_500;
 
 impl Server {
-    /// Proxy GET requests to /v0/order_book/* endpoints to the relayer
+    /// Handle a GET request to the /v0/order_book/:mint endpoint
+    ///
+    /// We deserialize the response from the relayer and replace the relayer fee
+    /// with the fee that the auth server will use for the given user.
     #[instrument(skip(self, path, headers))]
-    pub async fn handle_order_book_request(
+    pub async fn handle_order_book_request_with_mint(
+        &self,
+        mint: String,
+        path: warp::path::FullPath,
+        headers: HeaderMap,
+    ) -> Result<BytesResponse, Rejection> {
+        // Authorize the request
+        let path_str = path.as_str();
+        let (key_desc, key_id) = self
+            .authorize_request(path_str, "" /* query_str */, &headers, &[] /* body */)
+            .await?;
+
+        // Forward the request
+        let mut resp =
+            self.handle_order_book_request_internal(path_str, &key_desc, headers).await?;
```

Comment:
If the relayer returns an error, we should return it; currently we try to deserialize a Depth response which would return a confusing serde error to the user.